unit_test/blas/Test_Blas2_gemv_util.hpp

```diff
@@ -0,0 +1,343 @@
+#ifndef TEST_BLAS2_GEMV_UTIL_HPP
+#define TEST_BLAS2_GEMV_UTIL_HPP
+
+#include <gtest/gtest.h>
+#include <Kokkos_Core.hpp>
+#include <Kokkos_Random.hpp>
+#include <KokkosBatched_Vector_SIMD.hpp>
+
+namespace Test {
+
+template <typename ValueType, int length = KokkosBatched::DefaultVectorLength<
+                                  ValueType, TestExecSpace>::value>
+using simd_vector =
+    KokkosBatched::Vector<KokkosBatched::SIMD<ValueType>, length>;
+
+template <class AType, class XType, class YType, class ScalarType>
+struct GemvOpBase {
+  GemvOpBase(char trans_, ScalarType alpha_, AType A_, XType x_,
+             ScalarType beta_, YType y_)
+      : trans(trans_), alpha(alpha_), beta(beta_), A(A_), x(x_), y(y_) {}
+
+ protected:
+  // parameters
+  char trans;
+  ScalarType alpha;
+  ScalarType beta;
+  // data
+  AType A;
+  XType x;
+  YType y;
+};
+
+#define KK_DEFINE_BLAS2_GEMV_TEST_OP_CLASS(ClassName)                     \
+  struct ClassName : public GemvOpBase<AType, XType, YType, ScalarType> { \
+    using params = GemvOpBase<AType, XType, YType, ScalarType>;           \
+    ClassName(char trans_, ScalarType alpha_, AType A_, XType x_,         \
+              ScalarType beta_, YType y_)                                 \
+        : params(trans_, alpha_, A_, x_, beta_, y_) {}
+#define KK_END_BLAS2_GEMV_TEST_OP_CLASS \
```

Comment:
Not acceptable, this is very ugly code that is only needed because the previous macro is ill-defined.

Reply:
Thanks, lesson taken.